gon/contour.py

```diff
@@ -0,0 +1,148 @@
+from fractions import Fraction
+from functools import reduce
+from typing import (Iterable,
+                    List,
+                    Sequence)
+from weakref import WeakKeyDictionary
+
+from bentley_ottmann.planar import edges_intersect
+from memoir import cached
+from reprit.base import generate_repr
+from robust.hints import Expansion
+from robust.utils import (sum_expansions,
+                          two_product,
+                          two_two_diff)
+
+from gon.angular import (Orientation,
+                         to_orientation)
+from gon.geometry import Geometry
+from gon.hints import Coordinate
+from gon.point import (Point,
+                       RawPoint)
+
+MIN_VERTICES_COUNT = 3
+
+Vertices = Sequence[Point]
+RawContour = List[RawPoint]
+
+
+class Contour(Geometry):
+    __slots__ = '_vertices',
+
+    def __init__(self, vertices: Iterable[Point]) -> None:
+        self._vertices = tuple(vertices)
+
+    __repr__ = generate_repr(__init__)
+
+    def __hash__(self) -> int:
+        return hash(self._vertices)
+
+    def __eq__(self, other: 'Geometry') -> int:
+        return (self._vertices == other._vertices
+                if isinstance(other, Contour)
+                else NotImplemented)
+
+    @property
+    def vertices(self) -> Vertices:
+        return self._vertices
+
+    @cached.map_(WeakKeyDictionary())
+    def raw(self) -> RawContour:
+        return [vertex.raw() for vertex in self._vertices]
```

Comment:
One returns a tuple and another one returns a list. Is it fine?

Reply:
Maybe `_vertices` should be also a list since we work with homogeneous data.

Reply:
if `_vertices` will be a `list` it will be mutable and unhashable

Reply:
I don't see a problem with `_vertices` being mutable, but, regarding the second point, we could convert it to a tuple in `__hash__`.